tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/compose/HelmLegacyStatefulSetMigrationIT.java

```diff
@@ -0,0 +1,171 @@
+                WEB_DRIVER_CONTAINER,
+                CC_SERVICE_NAME,
+                CC_SERVICE_PORT);
+        // publish retained messages
+        final var topicList =
+                MqttUtil.publishRetainedMessages(client, operatorNamespace, MQTT_SERVICE_NAME, MQTT_SERVICE_PORT);
+        // assert RestAPI service and authorization with ESE extension
+        RestAPIUtil.assertAuth(client, operatorNamespace, REST_API_SERVICE_NAME, REST_API_SERVICE_PORT);
+        // assert publishes metrics
+        MonitoringUtil.assertPublishesMetrics(client, operatorNamespace, METRICS_SERVICE_NAME, METRICS_SERVICE_PORT);
+
+        // scale down the legacy operator
+        // kubectl scale deployment <release-name>-hivemq-operator-operator --replicas=0 -n <namespace>
+        final var legacyOperatorDeploymentName = String.format("%s-hivemq-operator-operator", LEGACY_RELEASE_NAME);
+        K8sUtil.scaleDeployment(client, operatorNamespace, legacyOperatorDeploymentName, 0);
+        var legacyOperatorLabels = K8sUtil.getHiveMQLegacyOperatorLabels(LEGACY_RELEASE_NAME);
+        client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .waitUntilCondition(Objects::isNull, 1, TimeUnit.MINUTES);
+        assertThat(client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .list()
+                .getItems()).isEmpty();
+
+        // delete custom resource with orphan
+        // kubectl delete hivemq-clusters.hivemq.com <release-name> --cascade=orphan -n <namespace>
+        final var legacyPlatform = K8sUtil.getLegacyHiveMQPlatform(client, operatorNamespace, LEGACY_RELEASE_NAME);
+        legacyPlatform.withPropagationPolicy(DeletionPropagation.ORPHAN).withTimeoutInMillis(5000).delete();
+        assertThat(K8sUtil.getLegacyHiveMQPlatform(client, operatorNamespace, LEGACY_RELEASE_NAME).get()).isNull();
+
+        // scale up the legacy operator
+        // kubectl scale deployment <release-name>-hivemq-operator-operator --replicas=1 -n <namespace>
+        K8sUtil.scaleDeployment(client, operatorNamespace, legacyOperatorDeploymentName, 1);
+        K8sUtil.waitForLegacyOperatorPodStateRunning(client, operatorNamespace, LEGACY_RELEASE_NAME);
+        assertThat(client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .list()
+                .getItems()
+                .size()).isEqualTo(1);
+
+        //TODO: We cannot execute `helm uninstall my-release --cascade orphan` - See: https://github.com/helm/helm/issues/13279
+        // we need to uninstall the legacy release with orphan option so the new platform Helm install takes over existing resources
+        //helmChartContainer.uninstallRelease(LEGACY_RELEASE_NAME, operatorNamespace, "--cascade", "orphan");
+
+        // as a workaround we need to delete the Helm secret installed
```

Comment:
A little bit of context would be great. I looked this up, and it seems like Helm v3 creates versioned secrets for every install, upgrade or roll back of a release. That should be the only management resource that is created. The naming schema should be sth. like `sh.helm.release.v1.<release-name>.v<release-version>`.

I guess this is also why you use a label matcher and not a name matcher, so it will remove all possible versions for the release.